File: Day172_2025/Q172_GFG.java
class Q172_GFG {
    public int catchThieves(char[] arr, int k) {
        int n = arr.length;
        int i = 0, j = 0, ans = 0;
        
        while (i < n && j < n) {
            if (arr[i] == 'T' && arr[j] == 'P') {
                if (Math.abs(i - j) <= k) {
                    ans++;
                    i++;
                    j++;
                } else {
                    if (i < j) i++;
                    else j++;
                }
            } else {
                while (i < n && arr[i] == 'P') i++;
                while (j < n && arr[j] == 'T') j++;
            }
        }

        return ans;
    }
}

File: Day172_2025/Q172_GFG.py
class Solution:
    def catchThieves(self, arr, k):
        n = len(arr)
        i, j = 0, 0
        ans = 0
        
        while i < n and j < n:
            if arr[i] == 'T' and arr[j] == 'P':
                if abs(i - j) <= k:
                    ans += 1
                    i += 1
                    j += 1
                else:
                    if i < j:
                        i += 1
                    else:
                        j += 1
            else:
                while i < n and arr[i] == 'P':
                    i += 1
                while j < n and arr[j] == 'T':
                    j += 1

        return ans
